src/hardware/voodoo.cpp

```diff
@@ -0,0 +1,7175 @@
+/*        Portion of this software comes with the following license:       */
+/***************************************************************************/
+/*
+    Copyright Aaron Giles
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are
+    met:
+
+        * Redistributions of source code must retain the above copyright
+          notice, this list of conditions and the following disclaimer.
+        * Redistributions in binary form must reproduce the above copyright
+          notice, this list of conditions and the following disclaimer in
+          the documentation and/or other materials provided with the
+          distribution.
+        * Neither the name 'MAME' nor the names of its contributors may be
+          used to endorse or promote products derived from this software
+          without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY AARON GILES ''AS IS'' AND ANY EXPRESS OR
+    IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+    DISCLAIMED. IN NO EVENT SHALL AARON GILES BE LIABLE FOR ANY DIRECT,
+    INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+    SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+    HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+    STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+    IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+
+***************************************************************************/
+
+/*
+  3dfx Voodoo Graphics SST-1/2 emulator by Aaron Giles
+  DOSBox integration by kekko (https://www.vogons.org/viewtopic.php?f=41&t=41853)
+  Code cleanups and multi-threaded triangle rendering by Bernhard Schelling
+*/
+
+#include "dosbox.h"
+
+#include <assert.h>
+#include <math.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <SDL.h>
+
+#if C_VOODOO
+#include "control.h"
+#include "cross.h"
+#include "mem.h"
+#include "paging.h"
+#include "pci_bus.h"
+#include "pic.h"
+#include "render.h"
+#include "setup.h"
+#include "vga.h"
+
+#ifndef DOSBOX_VOODOO_TYPES_H
+#define DOSBOX_VOODOO_TYPES_H
+
+/***************************************************************************
+    TYPE DEFINITIONS
+***************************************************************************/
+
+// FIXME Get rid of these typedefs
+//
+/* 8-bit values */
+typedef uint8_t UINT8;
+typedef int8_t INT8;
+
+/* 16-bit values */
+typedef uint16_t UINT16;
+typedef int16_t INT16;
+
+#ifndef _WINDOWS_
+/* 32-bit values */
+typedef uint32_t UINT32;
+typedef int32_t INT32;
+
+/* 64-bit values */
+typedef uint64_t UINT64;
+typedef int64_t INT64;
+#endif
```

Comment:
Nuke all typdefs and use `uint8_t` etc directly.

Reply:
I would love to, but... that would retouch more than 500 lines in voodoo.cpp, guaranteeing that we won't be able to apply improvements from other forks.

Opting to keep it as it is for now, but we'll gradually keep removing them as retouching lines (started already with e7995f6f1285f27c8723635dd47a8b467be770a1).